docs/runtimes/create-and-use-functions.ipynb

```diff
@@ -0,0 +1,550 @@
+    "```\n",
+    "\n",
+    "When creating a function, there are 3 main scenarios:\n",
+    "1. [**Single Source File**](#single-source-file) - when your code can be contained into a single file\n",
+    "2. [**Multiple Source Files**](#multiple-source-files) - when your code requires additional files or dependencies\n",
+    "3. [**Import Existing Function**](#import-existing-function) - when your function already exists elsewhere and you just want to import it\n",
+    "\n",
+    "``` {admonition} Note\n",
+    "Using the `set_function` method of an MLRun project allows for each of these scenarios in a transparent way. Depending on the source passed in, the project will register the function using some lower level functions. For specific use cases, you also have access to the lower level functions {py:func}`~mlrun.run.new_function`, {py:func}`~mlrun.run.code_to_function`, and {py:func}`~mlrun.run.import_function`.\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8ab1e780",
+   "metadata": {},
+   "source": [
+    "<a id=\"using-set_function\"></a>\n",
+    "### Using set_function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c878374d",
+   "metadata": {},
+   "source": [
+    "The MLRun project object has a method called {py:meth}`~mlrun.projects.MlrunProject.set_function` which is a one-size-fits-all way of creating an MLRun function. This method accepts a variety of sources including Python files, Jupyter notebooks, Git repos, and more.\n",
+    "\n",
+    "``` {admonition} Note\n",
+    "The return value of `set_function` will be your MLRun function. You can immediately run it or apply additional configuration like resources, scaling, etc. See [**Customizing Functions**](#customizing-functions) for more details.\n",
+    "```\n",
+    "\n",
+    "When using `set_function`, there are a number of common parameters across all function types and creation scenarios. Consider the following example:\n",
+    "\n",
+    "```python\n",
+    "fn = project.set_function(\n",
+    "    name=\"my-function\", tag=\"latest\", func=\"my_function.py\",\n",
+    "    image=\"mlrun/mlrun\", kind=\"job\", handler=\"train_model\",\n",
+    "    requirements=[\"pandas==1.3.5\"], with_repo=True\n",
+    ")\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bb9cebf9",
+   "metadata": {},
+   "source": [
+    "- **Name:** Name of your MLRun function within the given project. This will be displayed in the MLRun UI as well as the Kubernetes pod.\n",
+    "\n",
+    "- **Tag:** Tag for your function (much like a Docker image). Ommiting this parameter will default to `latest`.\n",
+    "\n",
+    "- **Func:** What to run with the MLRun function. This can be a number of things including files (`.py`, `.ipynb`, `.yaml`, etc.), URIs (`hub://` prefixed function marketplace URI, `db://` prefixed MLRun DB URI), existing MLRun Function objects, or `None` (for current `.ipynb` file).\n",
```

Comment:
"- **func:** What to run with the MLRun function. This can be a number of things including: files (`.py`, `.ipynb`, `.yaml`, etc.), URIs (`hub://` prefixed function marketplace URI, `db://` prefixed MLRun DB URI), existing MLRun Function objects, or `None` (for current `.ipynb` file).\n",